common/views/components/ButtonSolid/ButtonSolid.tsx

```diff
@@ -210,20 +218,29 @@ const Button: FC<ButtonSolidProps> = (
       ref={ref}
     >
       <BaseButtonInner>
-        <>
-          {icon && (
-            <ButtonIconWrapper>
-              <Icon icon={icon} />
-            </ButtonIconWrapper>
-          )}
+        {isIconAfter && (
+          <span
+            className={classNames({
+              'visually-hidden': !!isTextHidden,
+            })}
+          >
+            {text}
+          </span>
+        )}
+        {icon && (
```

Comment:
In doing it this way, is there a difference in margin (right/left) that takes place depending on where the icon is?

Reply:
Any margin is applied to the icon rather than the text, and the `iconAfter` prop on the `ButtonIconWrapper styled-component determines whether that margin is on the left or right of the icon https://github.com/wellcomecollection/wellcomecollection.org/blob/main/common/views/components/ButtonSolid/ButtonSolid.tsx#L96